jsonrpc/eth_endpoint.go

```diff
@@ -295,7 +298,20 @@ func (e *Eth) GetTransactionByHash(hash types.Hash) (interface{}, error) {
 func (e *Eth) GetTransactionReceipt(hash types.Hash) (interface{}, error) {
 	blockHash, ok := e.store.ReadTxLookup(hash)
 	if !ok {
-		// txn not found
+		// txn is not in any block
+		reason, err := e.store.GetTxDiscardReason(hash)
+		if err != nil {
+			e.logger.Warn(
+				"failed to get the reason of tx discard: hash=%s, err=%s",
+				hash,
+				err.Error(),
+			)
+		}
+
+		if reason != nil {
+			return *reason, nil
```

Comment:
This is not API-compatible with the JsonRPC spec, it might cause problems with the client implementation. I was under the impression that we would return it as an error of the JsonRPC call, which would be more compatible, and it kind of makes sense since this is an error on the transaction.

Reply:
Fixed 05377c9d21c5dc0241a8c9637eafdfc17b0b0bc9